music21/midi/translate.py

```diff
@@ -1860,91 +1868,73 @@ def midiTrackToStream(
-
-                # let tolerance for chord subbing follow the quantization
-                if quantizePost:
-                    divisor = max(quarterLengthDivisors)
-                # fallback: 1/16 of a quarter (64th)
-                else:
-                    divisor = 16
-                chunkTolerance = ticksPerQuarter / divisor
-                # must be strictly less than the quantization unit
-                if abs(tSub - t) < chunkTolerance:
-                    # isolate case where end time is not w/n tolerance
-                    if abs(tOffSub - tOff) > chunkTolerance:
-                        # need to store this as requiring movement to a diff
-                        # voice
-                        voicesRequired = True
-                        continue
-                    if chordSub is None:  # start a new one
-                        chordSub = [notes[i]]
-                        iGathered.append(i)
-                    chordSub.append(notes[j])
-                    iGathered.append(j)
-                    continue  # keep looping through events to see
-                    # if we can add more elements to this chord group
-                else:  # no more matches; assuming chordSub tones are contiguous
-                    break
-            # this comparison must be outside of j loop, as the case where we
-            # have the last note in a list of notes and the j loop does not
-            # execute; chordSub will be None
-            if chordSub is not None:
-                # composite.append(chordSub)
-                # create a chord here
-                c = chord.Chord()
-                midiEventsToChord(chordSub, ticksPerQuarter, c)
-                o = notes[i][0][0] / ticksPerQuarter
-                c.midiTickStart = notes[i][0][0]
-
-                s.coreInsert(o, c)
-                # iSkip = len(chordSub)  # amount of accumulated chords
-                chordSub = None
-            else:  # just append the note, chordSub is None
-                # composite.append(notes[i])
-                # create a note here
-                n = note.Note()
-                midiEventsToNote(notes[i], ticksPerQuarter, n)
-                # the time is the first value in the first pair
-                # need to round, as floating point error is likely
-                o = notes[i][0][0] / ticksPerQuarter
-                n.midiTickStart = notes[i][0][0]
-
-                s.coreInsert(o, n)
-                # iSkip = 1
-            # break  # exit secondary loop
-            i += 1
-
-    elif len(notes) == 1:  # rare case of just one note
-        n = note.Note()
-        midiEventsToNote(notes[0], ticksPerQuarter, n)
-        # the time is the first value in the first pair
-        # need to round, as floating point error is likely
-        o = notes[0][0][0] / ticksPerQuarter
-        n.midiTickStart = notes[i][0][0]
-        s.coreInsert(o, n)
+    for i, notePair in enumerate(notes):
```

Comment:
I'm not sure I understand why the special case of just one note can be removed?  I think we wouldn't have put it here if we didn't find it "in the wild"

Reply:
No reason to dwell on this, so I'll just restore it. But yes, it does the same thing as far as I can tell. My guess it was just caution about not choking on IndexError in `for j in range(i+1...`, maybe that was different in Python2?